### ISSUE TITLE
Tensorflow read a large number of images?

### ISSUE THREAD

**Issue by aureosun** [kwotsin/create_tfrecords#6]
Sir,I'm a beginner of deep learning and tensorflow , now I want to use the classification model I have trained on my own dataset to classify images which are all .jpg files,and the number of images I'd like to classify is about 3000,if I read and classify them one by one,the time is too long,do you have some suggestions about it?

**Comment by kwotsin** [comment]
If you are referring to an alternative method apart from what is provided in this repo, then I suggest you load your list of filenames like this as a Tensor and provide it to something like `slice_input_producer`.

Example:

```
        #Load the files into one input queue
        images = tf.convert_to_tensor(image_files)
        annotations = tf.convert_to_tensor(annotation_files)
        input_queue = tf.train.slice_input_producer([images, annotations]) #Slice_input producer shuffles the data by default.

        #Decode the image and annotation raw content
        image = tf.read_file(input_queue[0])
        image = tf.image.decode_image(image, channels=3)

        #preprocess and batch up the image and annotation
        preprocessed_image, preprocessed_annotation = preprocess(image, annotation, image_height, image_width)
        images, annotations = tf.train.batch([preprocessed_image, preprocessed_annotation], batch_size=batch_size, allow_smaller_final_batch=True)

```

Further example here: https://github.com/kwotsin/TensorFlow-ENet/blob/master/train_enet.py

except that in this case, your `annotation` is actually just a list of labels and you don't have to decode the annotation.

Hope it helps.
